src/pages/admin/AdminSessionPage.tsx

```diff
@@ -0,0 +1,217 @@
+import { useEffect, useState, FormEvent } from 'react';
+import DatePicker from 'react-datepicker';
+import 'react-datepicker/dist/react-datepicker.css';
+import {
+  Box,
+  Button,
+  VStack,
+  HStack,
+  Input,
+  Text,
+  Card,
+  CardBody,
+  CardHeader,
+  Heading,
+  FormControl,
+  FormLabel,
+  useColorModeValue,
+} from '@chakra-ui/react';
+import { ArrowBackIcon } from '@chakra-ui/icons';
+import { Calendar, Clock, Key, Users } from 'lucide-react';
+import { Link } from 'react-router-dom';
+import api from '../../services/api';
+
+interface AttendanceSession {
+  id: string;
+  title: string;
+  key: string;
+  expires: string;
+  attendees: string[];
+}
+
+const isSessionActive = (session: AttendanceSession): boolean => {
+  return new Date(session.expires) > new Date();
+};
+
+export default function AdminSessionPage() {
+  const [sessions, setSessions] = useState<AttendanceSession[]>([]);
+  const [loading, setLoading] = useState(true);
+  const [title, setTitle] = useState('');
+  const [key, setKey] = useState('');
+  const [expiresDate, setExpiresDate] = useState<Date | null>(null);
+  const [error, setError] = useState('');
+
+  const cardBg = useColorModeValue('white', 'gray.800');
+  const borderColor = useColorModeValue('gray.200', 'gray.700');
+
+  const activeSessions = sessions.filter(isSessionActive);
+
+  useEffect(() => {
+    fetchSessions();
+  }, []);
+
+  const fetchSessions = async () => {
+    try {
+      setLoading(true);
+      const response = await api.get('/engagement/attendance/');
+      setSessions(Array.isArray(response.data) ? response.data : []);
+    } catch (err) {
+      setError('Failed to fetch sessions');
+      console.error(err);
+      setSessions([]);
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  const handleSubmit = async (e: FormEvent) => {
+    e.preventDefault();
+    setError('');
+
+    if (!expiresDate) {
+      setError('Please select an expiration date');
+      return;
+    }
+
+    const expiresISO = expiresDate.toISOString();
+
+    try {
+      await api.post('/engagement/attendance/session', {
+        title,
+        key,
+        expires: expiresISO,
+      });
+
+      fetchSessions();
+      setTitle('');
+      setKey('');
+      setExpiresDate(null);
+    } catch (err: any) {
+      setError(err.response?.data?.error || 'Failed to create session');
+      console.error(err);
+    }
+  };
```

Comment:
Maybe create a custom hook for fetching sessions? It would clean the code up a good amount